### DISCUSSION TITLE
using "git push -u origin main" command failure

### DISCUSSION THREAD

**Discussion by AdhamElSamahy909** [community/community#43585] [Other Feature Feedback, Questions, & Ideas]
Every time I use the "git push -u origin main" command to push my files to the remote repositery , I get an error which says authentication failed after I enter my username and password.i generated a personal access token and tried entering it when asked for the password instead of my account's password but I got a message saying that "unable to access https://github/user
I literally do not know how to solve this!!!

**Comment by airtower-luna** [comment]
Can you show us the *exact* error message (you can censor the repository name, if it's private)? Without that it's hard to answer.

**Comment by hawkeye359** [comment]
Try using vscode built in terminal for pushing. It will ask for authentication through web which is comparatively easy. Or you can try the ssh.

**Reply by AdhamElSamahy909** [reply]
That worked!! Thanks a lot.

**Reply by hawkeye359** [reply]
please mark it as answer
